package.json

```diff
@@ -34,9 +34,11 @@
     "build:ui": "cd embark-ui && npm run build",
     "clean": "rimraf dist embark-*.tgz package embark-ui/build",
     "eslint": "eslint",
-    "install:core": "npm install",
-    "install:ui": "cd embark-ui && npm install",
-    "install_all": "npm-run-all install:*",
+    "fix-shrinkwrap": "replace \"http://\" \"https://\" ./npm-shrinkwrap.json",
+    "fix-web3": "npm un web3 && npm i web3@1.0.0-beta.34",
+    "install:core": "npm install && npm run fix-web3 && npm run fix-shrinkwrap",
+    "install:ui": "cd embark-ui && npm install && npm run fix-web3 && npm run fix-pkglock",
+    "install_all": "run-p install:*",
```

Comment:
Should we rename that one postinstall and have something like that:
```
  "postinstall:clean: "npm-run-all fix-web3 fix-shrinkwrap"
  "postinstall:ui": "cd embark-ui && npm install && npm run fix-web3 && npm run fix-pkglock"
  "postinstall": "run-p postinstall:*" 
```

I feel it makes more sense than having `install:core` that does an `npm install`

Reply:
It does make more sense to use `postinstall`, nice cath!

What if we skip the `:ui` and `:clean` variations of postinstall, and simply give embark's and embark-ui's each a `"postinstall": "npm-run-all fix-web3 fix-..` script? That way `install_all` can still be `"run-p install:*"`. The `install:core` takes longer, and in the meantime, because they're running in parallel, `install:ui` can move on to running it's fix scripts without waiting on `install:core` to finish.

Reply:
sounds good to me 👍

Reply:
I've made some changes:

`fix:shrinkwrap` and `fix:pkglock` are given in the `postshrinkwrap` script of embark and embark-ui, respectively. `postshrinkwrap` works for both `npm-shrinkwrap.json` and `package-lock.json`, even though that's not clearly indicated in the npm docs.

This allows the `https://` fixes to be applied any time `npm install` or `npm uninstall` is run, e.g. if a dev/dependency is being added or removed to embark/-ui not just when installing embark/-ui themselves. Also, they don't work correctly if they're included as part of `postinstall` because of a timing thing re: when `npm install` does a final adjustment to shrinkwrap / pkg lock following `postinstall`.

In addition to discovering that `fix:shrinkwrap/pkglock` don't work with `postinstall`, it turns out `postinstall` for the `fix:web3` scripts is a no-go, it's better to run them as the last tasks of `install:core` and `install:ui`. The reasons:

* If included in `postinstall` they'll run when the package is installed from the registry (I tested w/ an install from packed tarball). We don't want that — it's only important to do these fixes as part of our `release` logic, as part of doing `install_all` to make sure things are up-to-date before running the build, etc.
* It seems to break CI. This is really maddening 😡 , but if in the embark repo you do: `rm -rf node_modules && npm i` with `fix:web3` in `postinstall`, then you end up with undesired changes in the shrinkwrap and missing web3-related dependencies (sometimes?), bad enough that tests won't pass; I saw that happen on Travis. So the only way to be in a good state with node_modules and shrinkwrap is to already be in a good state 😱 😱 😱! When starting fresh (e.g. fresh clone from github) then things can get screwed up. I'm thinking a move to yarn is more and more a good idea, but I don't know how that can fully help unless we tell our users that installing with npm is unsupported because otherwise when doing releases we still need to make sure that shrinkwrap is in good shape. If the next release of web3 fixes these crazy dependency problems (which surely are related to some buggy npm behavior), then that will provide some relief too as we should be able to remove the `fix:web3` scripts.